### PR TITLE
Use #count instead of chaining methods

### PR DESCRIPTION
**Pull request by michaelstalker**

### CODE REVIEW

**Comment by mneumark** [comment]
If you have to perform a O(n) action in order to count, might as well use `#count` with a block instead of using two chained O(n) methods in a row.

File: ruby/README.md [hunk 1]
```diff
@@ -1770,6 +1770,29 @@ this rule only to arrays with two or more elements.
...
+    [1, 2, 3].reject { |e| e > 2 }.length
+    [1, 2, 3].select { |e| e > 2 }.count { |e| e.odd? }
+    [1, 2, 3].reject { |e| e > 2 }.count { |e| e.even? }
+    array.select(&:value).count
+
+    # good
+    [1, 2, 3].count { |e| e > 2 }
+    [1, 2, 3].count { |e| e < 2 }
+    [1, 2, 3].count { |e| e > 2 && e.odd? }
+    [1, 2, 3].count { |e| e < 2 && e.even? }
```
Comment (michaelstalker) [inline ruby/README.md]: Should this be `e <= 2`?

Comment (mneumark) [inline ruby/README.md]: I suppose it depends on what they are trying to do there. I think its suppose to be mostly meaningless code.

**Comment by michaelstalker** [comment]
Using `#count` with a block seems better than using `#select` or `#reject`, followed by `#size`. I kind of like the pipeline approach of using `#select` or `#reject` with a block that does one thing, followed by `#count` that has a block that does one thing. Clojure has a [threading macro](https://clojuredocs.org/clojure.core/-%3E) that does something like this.

File: ruby/README.md (hunk 1 above)
Comment (michaelstalker) [inline ruby/README.md]: I had thought `[1, 2, 3].count { |e| e < 2 && e.even? }` was supposed to provide an alternate implementation for `[1, 2, 3].reject { |e| e > 2 }.count { |e| e.even? }` above. If it's not, you can disregard my comment. :smile:

**Comment by lfender6445** [comment]
i'm fine either way on this but I'd say block followed by `{}.count` is more readable

**Pull request merged**